cluster/src/main/scala/kanaloa/reactive/dispatcher/ClusterAwareBackend.scala

```diff
@@ -0,0 +1,90 @@
+package kanaloa.reactive.dispatcher
+
+import java.util.UUID
+
+import akka.actor._
+import akka.cluster.routing.{ClusterRouterGroup, ClusterRouterGroupSettings}
+import akka.routing._
+import akka.pattern.ask
+import akka.util.Timeout
+import kanaloa.reactive.dispatcher.ClusterAwareBackend.RouteeRef
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
```

Comment:
Can you just provide a quick description of how this is meant to work, I think I follow, but want to make sure I understand all this.